### DISCUSSION TITLE
URL format for WSS

### DISCUSSION THREAD

**Discussion by liuyinze** [khoih-prog/WebSockets2_Generic#47] [Q&A]
Hi all,

I am using this package to trying out something. The format of URL WSS provided by the server to which I want to send the request is as following: _"wss://your.cumulocity.environment.fullqualifieddomainname/notification2/consumer/token=yourJwtTokenRequestedFromNotification2TokenService"_
However, in this library the format for using WSS is **client.connect("_wss://your-secured-server-ip:port/uri_");** So in this case how should I adapt my endpoint to the method defined in your library?

**Comment by khoih-prog** [comment]
> "wss://your.cumulocity.environment.fullqualifieddomainname/notification2/consumer/token=yourJwtTokenRequestedFromNotification2TokenService"

The port in **client.connect("wss://your-secured-server-ip:port/uri")** is optional, and port **443** will be used if not specified.

Try to paste the whole `url` to client.connect() and see if OK.

You certainly have to verify and test for yourself as your to-be-connected WSS server can behave differently, and without [**Minimal, Reproducible Example**](https://stackoverflow.com/help/minimal-reproducible-example), it's futile to guess what's going on, if failure.

Try to increase `_WEBSOCKETS_LOGLEVEL_` to max to debug by

```
#define _WEBSOCKETS_LOGLEVEL_     4
```

Good Luck,

**Reply by liuyinze** [reply]
Hi KH,

I tried with you proposal and it works. However, I still have another question and I raised it on another discussion session.

BRs,
YIL
